make/autoconf/lib-sleef.m4

```diff
@@ -0,0 +1,101 @@
+#
+# Copyright (c) 2023, Arm Limited. All rights reserved.
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+#
+# This code is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 only, as
+# published by the Free Software Foundation.  Oracle designates this
+# particular file as subject to the "Classpath" exception as provided
+# by Oracle in the LICENSE file that accompanied this code.
+#
+# This code is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# version 2 for more details (a copy is included in the LICENSE file that
+# accompanied this code).
+#
+# You should have received a copy of the GNU General Public License version
+# 2 along with this work; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+# or visit www.oracle.com if you need additional information or have any
+# questions.
+#
+
+###############################################################################
+#
+# Setup sleef framework
+#
+AC_DEFUN_ONCE([LIB_SETUP_SLEEF],
+[
+  AC_ARG_WITH(libsleef, [AS_HELP_STRING([--with-libsleef],
+      [specify prefix directory for the libsleef library
+      (expecting the libraries under PATH/lib and the headers under PATH/include)])])
+
+  ENABLE_LIBSLEEF=false
+  LIBSLEEF_CFLAGS=
+  LIBSLEEF_LIBS=
+
+  if test "x$NEEDS_LIB_SLEEF" = xfalse; then
+    if test "x${with_libsleef}" != "x" &&
+       test "x${with_libsleef}" != "xno"; then
+      AC_MSG_WARN([[libsleef is not used, so --with-libsleef is ignored]])
+    fi
+  else
+    LIBSLEEF_FOUND=no
+
+    if test "x${with_libsleef}" = "xno"; then
+      AC_MSG_NOTICE([libsleef is disabled])
+    else
+      if test "x${with_libsleef}" != "x" &&
+         test "x${with_libsleef}" != "xyes"; then
+        # Check the specified libsleef
+        AC_MSG_CHECKING([for the specified LIBSLEEF])
+        if test -e ${with_libsleef}/lib/libsleef.so &&
+           test -e ${with_libsleef}/include/sleef.h; then
```

Comment:
This fails on my system because libsleef is in `/usr/local/lib64/`. This is the correct place to look according to the Linux FHS. You should hard-code `/lib`

Reply:
Did you try to find the libsleef by passing `--with-libsleef=` ? Currently `--with-libsleef=` can only work for people manually built from sleef source code.

Reply:
Yes. It still failed.

Reply:
You need to expand this logic to cover more instances. See e.g. lib-ffi.m4 for inspiration.

Basic flow:
* if user has specified libsleef root with argument, check both lib/ and lib64/ under that root.
* if user has not specified libsleef root, and we have no SYSROOT, try PKG_CHECK
* Otherwise, look in well-known directories which is $SYSROOT/usr/[local/]lib[64].

Reply:
also, ideally, you will add the corresponding specific overrides like in ffi:
```
  AC_ARG_WITH(libffi-include, [AS_HELP_STRING([--with-libffi-include],
      [specify directory for the libffi include files])])
  AC_ARG_WITH(libffi-lib, [AS_HELP_STRING([--with-libffi-lib],
      [specify directory for the libffi library])])
```

Reply:
Thanks for the suggestion @magicus !

The check in current `lib-sleef.m4` is very common:

-  If user has specified libsleef root by '--with-libsleef', we assume it is the manually built sleef lib. So only `lib/` and `include/` is checked. And the flags are set based on that path.
- If user has not specified the libsleef root, and no `SYSROOT` is set, we try `PKG_CHECK` (like what you suggested)
- Otherwise, check `sleef.h`   
   - We assume the sleef module is installed under one of the valid system paths if the header can be found. So just linking with `-lsleef` will success.

It's an issue in current flow like what @theRealAph met. I will add the options like `--with-libsleef-lib` and `--with-libsleef-include` like ffi. Regarding to extending the check for`--with-libsleef`, I think we can just make it simple like what it is now. Or, we have to check all the potential valid lib paths like `lib/`, `lib64/`, or maybe `lib/aarch64-linux-gnu`. The same to the `include` part.  @theRealAph @magicus , WDYT?

Reply:
Added these two options in latest commit. Thanks!